### DISCUSSION TITLE
Does Watchtower update itself?

### DISCUSSION THREAD

**Discussion by sfkpmr** [containrrr/watchtower#565] [Q&A] [answered]
Does Watchtower update itself, or do I need to do it manually? I've set Watchtower to only update a couple of my containers, so would I need to add Watchtower to that same list of containers to update?

Just leaving it alone didn't seem to do anything, at least the container age didn't change.

Is it possible to check which version of Watchtower is currently running?

Installed like this:

`docker run -d \
    --name watchtower \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -e TZ=Europe/Stockholm \
    -e WATCHTOWER_SCHEDULE="0 0 6 * * *" \
    --restart unless-stopped \
    containrrr/watchtower \
    containername1 containername2 containername3`

**Comment by simonaronsson** [comment, marked as answer]
@victorcmoura elegantly explained how container selection works in https://github.com/containrrr/watchtower/discussions/519#discussioncomment-21473:
> Currently, to build a container filter, the following local filters are set as a kind of pipeline:
>
> - Filter by name (returns true for all containers that are on the names list if it is set)
> - Filter by enabled label (returns true for all containers with the label com.centurylinklabs.watchtower.enable=true if --label-enable is used)
> - Filter by disabled label (returns false for all containers with the label com.centurylinklabs.watchtower.enable=false, regardless of any argument)
>
> At runtime, every container that returns false for any one of these filters will not be monitored. So, if you don't include a container's name on the list (considering that there is a list), it should be ignored as it won't return true for this criteria.

Passing a list of containers to monitor, which does not include the watchtower container, will disable the monitoring of watchtower. By adding it to the argument list, it will start automatically updating itself.
